ggml-vulkan.cpp

```diff
@@ -734,17 +746,22 @@ static vk_buffer ggml_vk_create_buffer(ggml_backend_vk_context * ctx, size_t siz
 
     vk::PhysicalDeviceMemoryProperties mem_props = ctx->device.lock()->physical_device.getMemoryProperties();
 
-    uint32_t memory_type_index = UINT32_MAX;
+    uint32_t memory_type_index = -1;
```

Comment:
Basically your change behaves the same as it was before, but it hides an underflow. Was there a specific reason you preferred `-1` over `UINT32_MAX`? It'd only make sense if it was not unsigned, but I don't see an issue with `UINT32_MAX`, there will never be a device where that's a valid value.

Reply:
I was just using -1 as an error sentinel in find_properties, but forgot to update the sign. I'll revert to UINT32_MAX since that works too.